src/main/java/frc/robot/subsystems/targetvision/TargetVision.kt

```diff
@@ -1,62 +1,84 @@
-package frc.robot.subsystems.vision
+package frc.robot.subsystems.targetvision
 
 
 import edu.wpi.first.math.geometry.Translation2d
 import edu.wpi.first.math.util.Units
 import edu.wpi.first.wpilibj2.command.Subsystem
-import frc.robot.subsystems.drivetrain.Drivetrain
+import frc.robot.commands.AimingConstants
 import frc.robot.utils.LimelightHelpers.LimelightTarget_Retro
 import org.littletonrobotics.junction.Logger
 import kotlin.math.tan
 
-class Vision(private val io: VisionIO): Subsystem {
+object TargetVision:  Subsystem {
 
+    private val io = Limelight
     private val inputs: VisionIO.VisionIOInputs = VisionIO.VisionIOInputs()
 
+
+    fun getDistance(target: LimelightTarget_Retro): Double{
+        val angleToBucket = Units.degreesToRadians(LIMELIGHT_ANGLE + target.ty)
+        return (BUCKET_HEIGHT - LIMELIGHT_HEIGHT) / tan(angleToBucket)
+    }
+
+    //ascending
+    private val targetsbyDistance: List<Pair<LimelightTarget_Retro, Double>>
+        get() {
+            return inputs.targets.map {target: LimelightTarget_Retro -> Pair(target, getDistance(target))}.sortedBy {it.second}
+        }
+    val hasTargets: Boolean
+        get(){
+            return inputs.hasTargets
+        }
+    val curTime: Double
+        get() {return inputs.lastTimeStampMS}
```

Comment:
You should get the current time from NetworkTables rather than the time of the last update.